pubnub-chat-impl/src/jsMain/kotlin/ChannelJs.kt

```diff
@@ -0,0 +1,307 @@
+@file:OptIn(ExperimentalJsExport::class, ExperimentalJsStatic::class)
+
+import com.pubnub.chat.Channel
+import com.pubnub.chat.MessageDraft
+import com.pubnub.chat.internal.MessageDraftImpl
+import com.pubnub.chat.internal.channel.BaseChannel
+import com.pubnub.chat.types.ChannelType
+import com.pubnub.chat.types.InputFile
+import com.pubnub.chat.types.MessageMentionedUser
+import com.pubnub.chat.types.MessageReferencedChannel
+import com.pubnub.chat.types.TextLink
+import com.pubnub.kmp.JsMap
+import com.pubnub.kmp.UploadableImpl
+import com.pubnub.kmp.asFuture
+import com.pubnub.kmp.createJsObject
+import com.pubnub.kmp.then
+import com.pubnub.kmp.toJsMap
+import com.pubnub.kmp.toMap
+import kotlin.js.Json
+import kotlin.js.Promise
+import kotlin.js.json
+
+@JsExport
+@JsName("Channel")
+open class ChannelJs internal constructor(internal val channel: Channel) : ChannelFields {
+    override val id: String get() = channel.id
+    override val name: String? get() = channel.name
+    override val custom: Any? get() = channel.custom?.toJsMap() // todo recursive?
+    override val description: String? get() = channel.description
+    override val updated: String? get() = channel.updated
+    override val status: String? get() = channel.status
+    override val type: String? get() = channel.type?.stringValue
+
+    fun update(data: ChannelFields): Promise<ChannelJs> {
+        return channel.update(
+            data.name,
+            convertToCustomObject(data.custom), // TODO
+            data.description,
+            data.status,
+            ChannelType.from(data.type)
+        ).then {
+            it.asJs()
+        }.asPromise()
+    }
+
+    fun delete(options: DeleteParameters?): Promise<Any> {
+        return channel.delete(options?.soft ?: false).then {
+            it?.asJs() ?: true
```

Comment:
I'm making a change here to add type safety instead of `Any`, it will be something like `return DeleteResult(user)` or `return DeleteResult(true)` so it will be more readable that it's a result